app/src/main/java/com/github/se/wanderpals/model/viewmodel/OverviewViewModel.kt

```diff
@@ -41,4 +42,25 @@ open class OverviewViewModel(val tripsRepository: TripsRepository) : ViewModel()
       _isLoading.value = false
     }
   }
+  /**
+   * Adds the user to a trip with the specified trip ID. The trip must already exist in the database
+   * to make this function success.
+   *
+   * @param tripId The ID of the trip to join.
+   * @return True if the user successfully joins the trip, false otherwise.
+   */
+  open fun joinTrip(tripId: String): Boolean {
+    var success = false
```

Comment:
Maybe you can use lateinit in place of double initialisation

Reply:
I think that it is globably safer to let the success being false, so we can capture the logic that before trying the trip ID there is no success